cfme/tests/cloud_infra_common/test_html5_vm_console.py

```diff
@@ -93,80 +95,88 @@ def test_html5_vm_console(appliance, provider, configure_websocket, vm_obj,
+             flags=re.I | re.X)
+
+            def _validate_login():
+                """
+                Try to read what is on present on the last line in console.
+
+                If it is word 'login', enter username, if 'password' enter password, in order
+                to make the login successful
+                """
+                if vm_console.find_text_on_screen(text_to_find='login', current_line=True):
+                    vm_console.send_keys(console_vm_username)
+
+                if vm_console.find_text_on_screen(text_to_find='Password', current_line=True):
+                    vm_console.send_keys("{}\n".format(console_vm_password))
+                # if the login attempt failed for some reason (happens with RHOS-cirros),
+                # last line of the console will contain one of the following words:
+                # [login, password, incorrect]
+                # if so, regex_for_login_password will find it and result will not be []
+                # .split('\n')[-1] splits the console text on '\n' & picks last item of the list
+                result = regex_for_login_password.findall(vm_console.get_screen_text()
+                    .split('\n')[-1])
+                return result == []
+
+            # if _validate_login() returns True, it means we did not find any of words
+            # [login, password, incorrect] on last line of console text, which implies login success
+            wait_for(func=_validate_login, timeout=300, delay=5)
+
+            logger.info("Wait to get the '$' prompt")
+            if provider.one_of(VMwareProvider):
+                vm_console.wait_for_text(text_to_find=provider.data.templates
+                    .get('console_template')['prompt_text'], timeout=200)
+            else:
+                time.sleep(15)
+
+            # create file on system
+            vm_console.send_keys("touch blather")
+            if not (BZ.bugzilla.get_bug(1491387).is_opened):
+                # Test pressing ctrl-alt-delete...we should be able to get a new login prompt:
+                vm_console.send_ctrl_alt_delete()
+                assert vm_console.wait_for_text(text_to_find="login:", timeout=200,
+                    to_disappear=True), ("Text 'login:' never disappeared, indicating failure"
+                    " of CTRL+ALT+DEL button functionality, please check if OS reboots on "
+                    "CTRL+ALT+DEL key combination and CTRL+ALT+DEL button on HTML5 Console works.")
+                assert vm_console.wait_for_text(text_to_find="login:", timeout=200), ("VM Console"
+                    " didn't prompt for Login")
+
+            if not provider.one_of(OpenStackProvider):
+                assert vm_console.send_fullscreen(), ("VM Console Toggle Full Screen button does"
+                " not work")
+
+            # Ensure VM had pingable IP before attempting SSH
+            wait_pingable(vm_obj.mgmt, allow_ipv6=False, wait=240)
```

Comment:
We are restarting VM with Ctrl+Alt+Del and even after VM is restarted and prompts for `login:` , it does take some time before RHV can fetch the IP addr from VM and make it available via API to wrapanapi, hence if we don't `wait_pingable` then on next line, SSH Client fails to connect and fails the entire test. Hence adding this wait here is a must.